src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -19,6 +19,62 @@ namespace Azure.DataApiBuilder.Service.GraphQLBuilder.Sql
 {
     public static class SchemaConverter
     {
+        /// <summary>
+        /// Adds a field with roles and directives to the fields dictionary if the field is allowed to be part of the schema.
+        /// </summary>
+        /// <param name="rolesAllowedForFields">The dictionary that stores field names and their corresponding roles.</param>
+        /// <param name="fieldDefinitionName">The name of the field being processed.</param>
+        /// <param name="databaseObject">The database object representing the entity associated with the field.</param>
+        /// <param name="directives">A list of directives associated with the field.</param>
+        /// <param name="configEntity">The configuration entity for the field.</param>
+        /// <param name="fieldDefinition">The field definition object representing the field.</param>
+        /// <param name="fields">The dictionary that stores field names and their corresponding FieldDefinitionNode objects.</param>
+        private static void AddFieldWithRolesAndDirectives(
+            IDictionary<string, IEnumerable<string>> rolesAllowedForFields,
+            string fieldDefinitionName,
+            DatabaseObject databaseObject,
+            IList<DirectiveNode> directives,
+            [NotNull] Entity configEntity,
+            IFieldDefinition fieldDefinition,
+            IDictionary<string, FieldDefinitionNode> fields
+        ) {
+            // If no roles are allowed for the field, we should not include it in the schema.
+            // Consequently, the field is only added to schema if this conditional evaluates to TRUE.
+            if (rolesAllowedForFields.TryGetValue(key: fieldDefinitionName, out IEnumerable<string>? roles))
+            {
+                // Roles will not be null here if TryGetValue evaluates to true, so here we check if there are any roles to process.
+                // Since Stored-procedures only support 1 CRUD action, it's possible that stored-procedures might return some values
+                // during mutation operation (i.e, containing one of create/update/delete permission).
+                // Hence, this check is bypassed for stored-procedures.
+                if (roles.Count() > 0 || databaseObject.SourceType is SourceType.StoredProcedure)
```

Comment:
super nit: for consistency with other places in code, and since this is being modified, I would use `roles.Any()`. Even though the implementation for the IEnumerable is a list, another implementation may not use a collection which supports .count. .count is a short circuit within .Count() as seen [here, where worst case, the collection is fully enumerated to get the count. ](https://github.com/dotnet/runtime/blob/v6.0.16/src/libraries/System.Linq/src/System/Linq/Count.cs)